base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -252,6 +251,82 @@ where T: BlockchainBackend + 'static
                 }
                 Ok(NodeCommsResponse::HistoricalBlocks(blocks))
             },
+            NodeCommsRequest::FetchBlocksWithKernels(excess_sigs) => {
+                let mut blocks = Vec::<HistoricalBlock>::with_capacity(1);
+                for sig in excess_sigs {
+                    debug!(
+                        target: LOG_TARGET,
+                        "A peer has requested a block with kernel with sig {}",
+                        sig.get_signature().to_hex(),
+                    );
+                    match async_db::fetch_block_with_kernel(self.blockchain_db.clone(), sig.clone()).await {
+                        Ok(Some(block)) => blocks.push(block),
+                        Ok(None) => warn!(
+                            target: LOG_TARGET,
+                            "Could not provide requested block containing kernel with sig {} to peer because not \
+                             stored",
+                            sig.get_signature().to_hex(),
+                        ),
+                        Err(e) => warn!(
+                            target: LOG_TARGET,
+                            "Could not provide requested block containing kernel with sig {} to peer because: {}",
+                            sig.get_signature().to_hex(),
+                            e.to_string()
+                        ),
```

Comment:
Should this error not get sent back to the client?

Reply:
I kept it the same as all the other get_block function. 

They all return a vec<block>. This might only relate to a single block failing and the rest being fine.